client-misk/src/main/kotlin/app/cash/backfila/client/misk/HibernateBackfilaClientModule.kt

```diff
@@ -0,0 +1,42 @@
+package app.cash.backfila.client.misk
+
+import app.cash.backfila.client.misk.internal.BackfillOperator
+import app.cash.backfila.client.misk.internal.HibernateBackfillOperator
+import com.google.inject.Binder
+import com.google.inject.TypeLiteral
+import com.google.inject.multibindings.MapBinder
+import kotlin.reflect.KClass
+import kotlin.reflect.jvm.jvmName
+import misk.inject.KAbstractModule
+
+class HibernateBackfilaClientModule() : KAbstractModule() {
+  override fun configure() {
+    multibind<BackfillOperator.Backend>().to<HibernateBackfillOperator.HibernateBackend>()
+
+    mapBinder(binder()) // Default the mapbinder to having no Backfills registered.
+  }
+}
+
+class HibernateBackfillInstallModule<T : Backfill<*, *, *>> private constructor(
```

Comment:
I guess I don't agree with this degree of separation - are  we going to duplicate this for each base type of backfill? could we not have one Backfill interface that they all implement?

Reply:
I think this is a reasonable thing to duplicate since the backend might have some other way of specifying the backfills. It allows them to diverge a bit more so we avoid locking them all together. Makes it a has-a rather than an is-a.

You could have a backend that generates the backfills.

For instance we may for dynamo we might always want to add certain parameters. This allows us to do this in many ways rather than being forced to have a super Parameters type.

Reply:
I think the implementations will all require massive changes to migrate a backfill between them. I think this is ok and actually what we want customers to do.